library/Test/Tasty/Generator.hs

```diff
@@ -134,17 +135,26 @@ hspecTestCaseGenerator = Generator
   , generatorSetup  = \t -> "HS.testSpec \"" ++ name t ++ "\" " ++ qualifyFunction t
   }
 
+-- | Custom test generator prefix.
+customGroupGenerator :: Generator
+customGroupGenerator = Generator
+  { generatorPrefix = "custom_"
+  , generatorImport = "import qualified Test.Tasty.Discover.Custom as TD\n"
+  , generatorClass  = ""
+  , generatorSetup  = \t -> "TD.fromCustomTest \"" ++ name t ++ "\" " ++ qualifyFunction t
+  }
+
 -- | Tasty group generator prefix.
 tastyTestGroupGenerator :: Generator
 tastyTestGroupGenerator = Generator
   { generatorPrefix = "test_"
   , generatorImport = ""
   , generatorClass  = concat
-    [ "class TestGroup a where testGroup :: String -> a -> IO T.TestTree\n"
-    , "instance TestGroup T.TestTree        where testGroup _ a = pure a\n"
-    , "instance TestGroup [T.TestTree]      where testGroup n a = pure $ T.testGroup n a\n"
-    , "instance TestGroup (IO T.TestTree)   where testGroup _ a = a\n"
-    , "instance TestGroup (IO [T.TestTree]) where testGroup n a = T.testGroup n <$> a\n"
+    [ "class ToTestTree a where toTestTree :: String -> a -> IO T.TestTree\n"
+    , "instance ToTestTree T.TestTree        where toTestTree _ a = pure a\n"
+    , "instance ToTestTree [T.TestTree]      where toTestTree n a = pure $ T.testGroup n a\n"
+    , "instance ToTestTree (IO T.TestTree)   where toTestTree _ a = a\n"
+    , "instance ToTestTree (IO [T.TestTree]) where toTestTree n a = T.testGroup n <$> a\n"
```

Comment:
This looks very similar to `FromCustomTest`.  How are they different?

Reply:
I mentioned this in the PR description:

> Ideally, we would reuse the `test_` prefix (which is why I made a small rename change) so that `TestTree`, `[TestTree]`, `IO TestTree`, and `IO [TestTree]` are just out-of-the-box implementations for an extendable `ToTestTree` class, but that would be a breaking change, forcing anyone using `test_` to specify `tasty-discover` as a dependency. Personally, I would be okay with it (and if maintainers are ok with it, I would prefer making this change), but I'd rather get the functionality merged soon than have the functionality blocked on reluctance to make a breaking change.

Basically, the `ToTestTree` type class is defined in-line here, so users can't create instances for this exact type class. In order to do so, we'd have to define (and export) `ToTestTree` from `tasty-discover`, which is a breaking change, since users were not required to be able to import from `tasty-discover` up til now